reference/vane-apis/ames.md

```diff
@@ -0,0 +1,303 @@
++++
+title = "Ames Public API"
+weight = 1
+template = "doc.html"
++++
+
+# Ames
+
+In this document we describe the public interface for Ames.  Namely, we describe
+each `task` that Ames can be `%pass`ed, and which `gift`(s) Ames can `%give` in return.
+
+Some `task`s appear to have more than one arm associated to them, e.g. there are
+four `+on-hear` arms. We denote this where it occurs, but always refer to the
+`on-hear:event-core` arm.
+
+Ames `task`s can be naturally divided into two categories: messaging tasks and
+system/lifecycle tasks.
+
+## Messaging Tasks
+
+### %hear
+
+`%hear` handles raw packet receipt. This `task` only ever originates from Unix.
+It does the initial processing of a packet, namely by passing the raw packet
+information to `+decode-packet` which deserializes the packet and giving that data and the origin of the
+packet to `+on-hear-packet`.
+
+There are multiple `+on-hear` arms in `ames.hoon`. Here we refer to
+`on-hear:event-core`, as that is the one called by a `%hear` `task`. The other ones are used
+primarily for ack and nack processing, or receiving message fragments.
+
+#### Accepts
+ 
+```hoon
+[=lane =blob]
+```
+
+`%hear` takes in a `blob`, which is essentially a large atom (around 1kB or less)
+that is the raw data of the message, and a `lane`, which is the origin of the
+message (typically an IP address).
+
+#### Returns
+
+`%hear` can trigger any number of possible returns...?
```

Comment:
Yep, depending on many factors (esp. congestion control), it can cause 0 or more additional packets to be released.  It may also trigger a `%memo` or `%boon` to a vane (if it completes a message).